src/micro_http/src/common/headers.rs

```diff
@@ -199,6 +205,12 @@ impl Headers {
                             )),
                         },
                         Header::Server => Ok(()),
+                        Header::AcceptEncoding => Encoding::try_from(entry[1].trim().as_bytes())
+                            .map(|_| {
+                                self.accept_encoding = true;
+                                Ok(())
+                            })
+                            .map_err(|e| e)?,
```

Comment:
If you drop the previous `accept_encoding` field, which is unused later anyway, this becomes much simpler.
```suggestion
                        Header::AcceptEncoding => Encoding::try_from(entry[1].trim().as_bytes()),
```